modules/imgproc/include/opencv2/imgproc.hpp

```diff
@@ -1536,12 +1536,13 @@ respectively (see #getGaussianKernel for details); to fully control the result r
 possible future modifications of all this semantics, it is recommended to specify all of ksize,
 sigmaX, and sigmaY.
 @param borderType pixel extrapolation method, see #BorderTypes. #BORDER_WRAP is not supported.
+@param hint Implementation modfication flags. See #ImplementationHint
 
 @sa  sepFilter2D, filter2D, blur, boxFilter, bilateralFilter, medianBlur
  */
 CV_EXPORTS_W void GaussianBlur( InputArray src, OutputArray dst, Size ksize,
                                 double sigmaX, double sigmaY = 0,
-                                int borderType = BORDER_DEFAULT );
+                                int borderType = BORDER_DEFAULT, int hint = IMPL_DEFAULT );
```

Comment:
use explicit type instead of int

Reply:
what's the meaning of Fast+Inherit? it's very confusing. In my opinion, inherit is bad by design.

It should be explicit pass of algorithm hint from the parent function to child:

```
void opticalFlowABC(...., AlgoHint hint) {

      GaussianBlur(prev_layer, curr_layer, ...., hint); // pass hint explicitly from the parent
}
```